index.js

```diff
@@ -343,163 +341,159 @@ Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, ou
-    let outputPath = filter.getDestFilePath(relativePath, entry);
+      if (outputPath == null) {
+        throw new Error('canProcessFile("' + relativePath +
+                        '") is true, but getDestFilePath("' +
+                        relativePath + '") is null');
+      }
 
-    if (outputPath == null) {
-      throw new Error('canProcessFile("' + relativePath +
-                      '") is true, but getDestFilePath("' +
-                      relativePath + '") is null');
-    }
+      outputPath = destDir + '/' + outputPath;
 
-    outputPath = destDir + '/' + outputPath;
+      if (isChange) {
+        let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
+        if (isSame) {
+          this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
+          return;
+        } else {
+          this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+        }
+      }
 
-    if (isChange) {
-      let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
-      if (isSame) {
-        this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
-        return;
-      } else {
-        this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+      try {
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
+
+      } catch(e) {
+        // optimistically assume the DIR was patched correctly
+        mkdirp.sync(path.dirname(outputPath));
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
       }
-    }
 
-    try {
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
+      return outputString;
+    });
+  }
 
-    } catch(e) {
-      // optimistically assume the DIR was patched correctly
-      mkdirp.sync(path.dirname(outputPath));
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
-    }
+  /**
+   * @param contents {string}
+   * @param relativePath {string}
+   * @returns {string}
+   */
+  processString (contents, relativePath) { // jshint ignore:line
+    throw new Error(
+        'When subclassing broccoli-persistent-filter you must implement the ' +
+        '`processString()` method.');
+  }
 
-    return outputString;
-  });
-};
+  postProcess (result /*, relativePath */) {
+    return result;
+  }
+}
 
-/**
- * @param contents {string}
- * @param relativePath {string}
- * @returns {string}
- */
-Filter.prototype.processString = function(contents, relativePath) { // jshint ignore:line
-  throw new Error(
-      'When subclassing broccoli-persistent-filter you must implement the ' +
-      '`processString()` method.');
-};
-
-Filter.prototype.postProcess = function(result /*, relativePath */) {
-  return result;
-};
+module.exports = Filter;
```

Comment:
Can we do this in one expression above? This ensures as one reads, it is totally obvious what is being exported. Once we can switch to ES6 modules, this will likely eventually become `export default class PersistentFilter extends Plugin { };`

```js
// use strict;

module.exports = class PersistentFilter extends Plugin {

};
```

Reply:
Done